src/components/Editor/SearchBar.js

```diff
@@ -266,41 +304,67 @@ const SearchBar = React.createClass({
     this.traverseResults(e, e.shiftKey);
   },
 
-  renderSummary() {
-    const { searchResults: { count, index }} = this.props;
+  onKeyDown(e: SyntheticKeyboardEvent) {
+    if (!this.state.functionSearchEnabled) {
+      e.preventDefault();
+      return;
+    }
+    const searchResults = this.getFunctionResults(),
+      resultCount = searchResults.length;
+
+    if (e.key === "ArrowUp") {
+      const selectedResultIndex = Math
+        .max(0, this.state.selectedResultIndex - 1);
+      this.setState({ selectedResultIndex });
+      this.onSelectResultItem(searchResults[selectedResultIndex]);
+      e.preventDefault();
+    } else if (e.key === "ArrowDown") {
+      const selectedResultIndex = Math
+        .min(resultCount - 1, this.state.selectedResultIndex + 1);
+      this.setState({ selectedResultIndex });
+      this.onSelectResultItem(searchResults[selectedResultIndex]);
+      e.preventDefault();
+    } else if (e.key === "Enter") {
+      if (searchResults.length) {
+        this.selectResultItem(searchResults[this.state.selectedResultIndex]);
+      } else {
+        this.closeSearch(e);
+      }
+      e.preventDefault();
+    } else if (e.key === "Tab") {
+      this.closeSearch(e);
+      e.preventDefault();
+    }
+  },
+
+  // Renderers
+  buildSummaryMsg() {
+    const {
+      searchResults: { count, index },
+      query
+    } = this.props;
 
-    if (this.props.query.trim() == "") {
-      return dom.div({});
+    if (query.trim() == "") {
+      return "";
     }
 
     if (count == 0) {
-      return dom.div(
-        { className: "summary" },
-        L10N.getStr("editor.noResults")
-      );
+      return L10N.getStr("editor.noResults");
     }
 
     if (index == -1) {
-      return dom.div(
-        { className: "summary" },
-        L10N.getFormatStr("sourceSearch.resultsSummary1", count)
-      );
+      return L10N.getFormatStr("sourceSearch.resultsSummary1", count);
     }
 
-    return dom.div(
-      { className: "summary" },
-      L10N.getFormatStr("editor.searchResults", index + 1, count)
-    );
+    return L10N.getFormatStr("editor.searchResults", index + 1, count);
   },
 
-  renderSvg() {
-    const { searchResults: { count }} = this.props;
-
-    if (count == 0 && this.props.query.trim() != "") {
-      return Svg("sad-face");
+  buildPlaceHolder() {
```

Comment:
Not sure it's necessary to change the placeholder but thought indicating the search type was good.